cpp/bridges_module/algorithm/bridges.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <mg_graph.hpp>
+
+namespace bridges_util {
+
+/// Simple struct that keeps the state of nodes in algorithms that rely on
+/// DFS traversal.
+struct NodeState {
+  std::vector<bool> visited;
+  std::vector<std::uint64_t> discovery, low_link, depth;
+  std::vector<std::uint64_t> parent;
+  std::uint64_t counter;
+
+  explicit NodeState(std::uint64_t number_of_nodes) {
+    visited.resize(number_of_nodes, false);
+    discovery.resize(number_of_nodes, 0);
+    low_link.resize(number_of_nodes, 0);
+    counter = 0;
+  }
+
+  void Update(std::uint64_t node_id) {
+    counter++;
+    visited[node_id] = true;
+    discovery[node_id] = counter;
+    low_link[node_id] = counter;
+  }
+};
```

Comment:
This can be put into utils now right?

Reply:
Not sure that this is useful for each algorithm, only ones that are copied from previous implementation repository